app/dlc-suredbits-client/src/main/scala/org/bitcoins/sbclient/Client.scala

```diff
@@ -0,0 +1,130 @@
+package org.bitcoins.sbclient
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{HttpRequest, Uri}
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
+import org.bitcoins.core.crypto.{
+  AesCrypt,
+  AesEncryptedData,
+  AesKey,
+  ECPublicKey,
+  SchnorrDigitalSignature
+}
+import org.bitcoins.core.protocol.ln.{LnInvoice, PaymentPreimage}
+import org.bitcoins.eclair.rpc.api.{EclairApi, OutgoingPaymentStatus}
+import play.api.libs.json.{JsError, JsSuccess, JsValue, Json, Reads}
+
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutor, Future}
+import scala.concurrent.duration.DurationInt
+
+object Client {
+
+  def rawRestCall(uri: Uri)(implicit system: ActorSystem): Future[String] = {
+    implicit val m: ActorMaterializer = ActorMaterializer.create(system)
+    implicit val ec: ExecutionContextExecutor = m.executionContext
+
+    Http()
+      .singleRequest(HttpRequest(uri = uri))
+      .flatMap(
+        response =>
+          response.entity.dataBytes
+            .runFold(ByteString.empty)(_ ++ _)
+            .map(payload => payload.decodeString(ByteString.UTF_8))
+      )
+  }
+
+  def restCall(uri: Uri)(implicit system: ActorSystem): Future[JsValue] = {
+    rawRestCall(uri).map(Json.parse)(system.dispatcher)
+  }
+
+  import org.bitcoins.eclair.rpc.client.JsonReaders.lnInvoiceReads
+  implicit val invoiceAndDataResponseReads: Reads[InvoiceAndDataResponse] =
+    Json.reads[InvoiceAndDataResponse]
+
+  def request(
+      exchange: Exchange,
+      tradingPair: TradingPair,
+      requestType: RequestType)(
+      implicit system: ActorSystem): Future[InvoiceAndDataResponse] = {
+    implicit val ec: ExecutionContextExecutor = system.dispatcher
+
+    val prefix = "https://test.api.suredbits.com/dlc/v0"
```

Comment:
This should probably be part of the config or at least defined somewhere higher up

Reply:
It could be passed as a constructor parameter along with the `ActorSystem`. The later would help to avoid issues to mixed up actor systems and/or execution context.